ReactiveCocoa/Swift/Signal.swift

```diff
@@ -139,6 +143,12 @@ public final class Signal<Value, Error: ErrorType> {
 			return nil
 		}
 	}
+	
+	deinit {
+		if atomicObservers.value?.count ?? 0 == 0 {
```

Comment:
I'm confused about this. Can you explain why you'd only dispose if there are no observers?